### ISSUE TITLE
Building tests not successful on Mac OSX 

### ISSUE THREAD

**Issue by Suryamadhan** [dotnet/corefx#22177]
Hi,
I was trying to build core fx tests on my Macbook with OSX 10.12.3 and it failed to build the tests. 

I cloned corefx yesterday, I was able to build successfully. But when I built and ran the test using 
build-tests.sh I ran into errors. 

Here is the test execution summary. I have also attached the relevant files.

[Archive.zip](https://github.com/dotnet/corefx/files/1143416/Archive.zip)

```
/Users/Surya/code/dotnet2/corefx/Tools/tests.targets(346,5): warning :    System.Net.NameResolution.Pal.Tests  Total: 17, Errors: 0, Failed: 2, Skipped: 0, Time: 17.235s [/Users/Surya/code/dotnet2/corefx/src/System.Net.NameResolution/tests/PalTests/System.Net.NameResolution.Pal.Tests.csproj]
  ~/code/dotnet2/corefx/src/System.Net.NameResolution/tests/PalTests
  Finished running tests. End time=11:11:21. Return value was 1
/Users/Surya/code/dotnet2/corefx/Tools/tests.targets(346,5): warning MSB3073: The command "/Users/Surya/code/dotnet2/corefx/bin/Unix.AnyCPU.Debug/System.Net.NameResolution.Pal.Tests/netcoreapp/RunTests.sh /Users/Surya/code/dotnet2/corefx/bin/testhost/netcoreapp-OSX-Debug-x64/" exited with code 1. [/Users/Surya/code/dotnet2/corefx/src/System.Net.NameResolution/tests/PalTests/System.Net.NameResolution.Pal.Tests.csproj]
/Users/Surya/code/dotnet2/corefx/Tools/tests.targets(354,5): error : One or more tests failed while running tests from 'System.Net.NameResolution.Pal.Tests' please check /Users/Surya/code/dotnet2/corefx/bin/Unix.AnyCPU.Debug/System.Net.NameResolution.Pal.Tests/netcoreapp/testResults.xml for details! [/Users/Surya/code/dotnet2/corefx/src/System.Net.NameResolution/tests/PalTests/System.Net.NameResolution.Pal.Tests.csproj]
/Users/Surya/code/dotnet2/corefx/dir.traversal.targets(77,5): error : (No message specified) [/Users/Surya/code/dotnet2/corefx/src/tests.builds]

Build FAILED.

/Users/Surya/code/dotnet2/corefx/Tools/tests.targets(346,5): warning :    System.Net.WebProxy.Tests  Total: 21, Errors: 0, Failed: 1, Skipped: 0, Time: 5.925s [/Users/Surya/code/dotnet2/corefx/src/System.Net.WebProxy/tests/System.Net.WebProxy.Tests.csproj]
/Users/Surya/code/dotnet2/corefx/Tools/tests.targets(346,5): warning MSB3073: The command "/Users/Surya/code/dotnet2/corefx/bin/AnyOS.AnyCPU.Debug/System.Net.WebProxy.Tests/netstandard/RunTests.sh /Users/Surya/code/dotnet2/corefx/bin/testhost/netcoreapp-OSX-Debug-x64/" exited with code 1. [/Users/Surya/code/dotnet2/corefx/src/System.Net.WebProxy/tests/System.Net.WebProxy.Tests.csproj]
/Users/Surya/code/dotnet2/corefx/Tools/tests.targets(346,5): warning :    System.Net.NameResolution.Pal.Tests  Total: 17, Errors: 0, Failed: 2, Skipped: 0, Time: 17.235s [/Users/Surya/code/dotnet2/corefx/src/System.Net.NameResolution/tests/PalTests/System.Net.NameResolution.Pal.Tests.csproj]
/Users/Surya/code/dotnet2/corefx/Tools/tests.targets(346,5): warning MSB3073: The command "/Users/Surya/code/dotnet2/corefx/bin/Unix.AnyCPU.Debug/System.Net.NameResolution.Pal.Tests/netcoreapp/RunTests.sh /Users/Surya/code/dotnet2/corefx/bin/testhost/netcoreapp-OSX-Debug-x64/" exited with code 1. [/Users/Surya/code/dotnet2/corefx/src/System.Net.NameResolution/tests/PalTests/System.Net.NameResolution.Pal.Tests.csproj]
/Users/Surya/code/dotnet2/corefx/Tools/tests.targets(354,5): error : One or more tests failed while running tests from 'System.Net.WebProxy.Tests' please check /Users/Surya/code/dotnet2/corefx/bin/AnyOS.AnyCPU.Debug/System.Net.WebProxy.Tests/netstandard/testResults.xml for details! [/Users/Surya/code/dotnet2/corefx/src/System.Net.WebProxy/tests/System.Net.WebProxy.Tests.csproj]
/Users/Surya/code/dotnet2/corefx/Tools/tests.targets(354,5): error : One or more tests failed while running tests from 'System.Net.NameResolution.Pal.Tests' please check /Users/Surya/code/dotnet2/corefx/bin/Unix.AnyCPU.Debug/System.Net.NameResolution.Pal.Tests/netcoreapp/testResults.xml for details! [/Users/Surya/code/dotnet2/corefx/src/System.Net.NameResolution/tests/PalTests/System.Net.NameResolution.Pal.Tests.csproj]
/Users/Surya/code/dotnet2/corefx/dir.traversal.targets(77,5): error : (No message specified) [/Users/Surya/code/dotnet2/corefx/src/tests.builds]
    4 Warning(s)
    3 Error(s)

Time Elapsed 00:03:42.99
Command execution failed with exit code 1.
```

**Comment by pjanotti** [comment]
@Priya91 this is a repro of #13309. Initially I was able to repro it on my Mac and I noticed that the returned by `Dns.GetHostName()` had a suffix `".local"`. The tests passed when I hard coded the name of the machine without the suffix. The strange part is that after I experimented calling `gethostname` from libc and some other steps I lost the repro. I am going to try to clean up to see if I can get a consistent report.
